Flat/interface/TemplatedPandaAnalyzer.h

```diff
@@ -192,35 +201,46 @@ void PandaAnalyzer::FillGenTree(panda::Collection<T>& genParticles)
         break;
       }
     }
-    if (!jetOverlapsLepton) {
+    if (jetOverlapsLepton) 
+      continue;
+    if (!matchTarget) {
       fullJet = &testJet;
       break;
+    } else {
+      float matchDeltaR = DeltaR2(testJet.eta(), testJet.phi(), matchTarget->eta(), matchTarget->phi());
+      float matchPtBalance = fabs(testJet.pt() - matchTarget->pt()) / testJet.pt();
+      if (matchDeltaR < dR2 && matchPtBalance < 0.5) {
+        fullJet = &testJet;
+        break;
+      }       
     }
   }
-
-  tr->TriggerSubEvent("clustering gen");
-
   if (fullJet == NULL) {
+    PDebug("PandaAnalyzer::GenJetProperties","No gen jet found, exiting");
```

Comment:
- It's returning, not exiting. Those mean very different things
- There are plenty of events where you don't have a gen jet passing our various stringent requirements. No need to have a debug statement for all of them, unless you want 10s of GBs of log files.